react-common/components/profile/BadgeInfo.tsx

```diff
@@ -15,24 +15,24 @@ export const BadgeInfo = (props: BadgeInfoProps) => {
         <div className="profile-badge-info-image">
             <Badge badge={badge} disabled={!badge.timestamp} />
         </div>
-        <div className="profile-badge-info-header">
-            {lf("Awarded For:")}
-        </div>
-        <div className="profile-badge-info-text">
-            {badgeDescription(badge)}
-        </div>
-        { badge.timestamp ?
+        <div className="profile-badge-info-item">
             <div className="profile-badge-info-header">
-                {lf("Awarded On:")}
+                {lf("Awarded For:")}
             </div>
-            : undefined
-        }
-        { badge.timestamp ?
             <div className="profile-badge-info-text">
-                {date.toLocaleDateString(pxt.U.userLanguage())}
+                {badgeDescription(badge)}
             </div>
-            : undefined
-        }
+        </div>
+        { badge.timestamp ?
```

Comment:
small: `badge.timestamp &&` ?

Reply:
timestamp is a number, so if it's zero it results in a 0 being printed. I think I was initializing it to 0 at some point?

Reply:
could do !== undefined or something but it's also not a huge deal